node/rpc-api/build.rs

```diff
@@ -0,0 +1,522 @@
+            ));
+            method_table.push(format!(
+                "| - | {} | {} | {}",
+                method_pad_dash, params_pad_dash, result_pad_dash
+            ));
+
+            for (lotus_name, lotus_method) in lotus_rpc.iter() {
+                let forest_method = forest_rpc.get(lotus_name);
+
+                let status = match forest_method {
+                    Some(_method) => "✔️",
+                    None => "❌",
+                };
+
+                let (forest_params, forest_result) = match forest_method {
+                    Some(method) => (
+                        format!("({})", method.params.join(", ")),
+                        method.result.clone(),
+                    ),
+                    None => ("-".to_owned(), "-".to_owned()),
+                };
+
+                // Pad strings for display
+                let method_pad = " ".repeat(longest_method_name_len - lotus_method.name.len());
+
+                method_table.push(format!(
+                    "{} | `{}`{} | `{}` | `{}` |",
+                    status, lotus_method.name, method_pad, forest_params, forest_result,
+                ));
+            }
+
+            let forest_count = forest_rpc.len();
+            let lotus_count = lotus_rpc.len();
+            let api_coverage = (forest_count as f32 / lotus_count as f32) * 100.0;
+
+            let params_mismatches_table = params_mismatches
+                .iter()
+                .map(|(method, param_index, forest_param, lotus_param)| {
+                    format!(
+                        "| `{method}`{method_space} | `{param_index}` | `{forest_param}` | `{lotus_param}`",
+                        method = method,
+                        method_space = " ".repeat(longest_method_name_len - method.len()),
+                        param_index = param_index,
+                        forest_param = forest_param,
+                        lotus_param = lotus_param
+                    )
+                })
+                .collect::<Vec<String>>();
+
+            let result_mismatches_list = result_mismatches
+                .iter()
+                .map(|(method, forest_result, lotus_result)| {
+                    format!(
+                        "| `{method}`{method_space} | `{forest_result}` | `{lotus_result}`",
+                        method = method,
+                        method_space = " ".repeat(longest_method_name_len - method.len()),
+                        forest_result = forest_result,
+                        lotus_result = lotus_result
+                    )
+                })
+                .collect::<Vec<String>>();
+
+            let forest_only_methods_list = forest_only_methods
+                .iter()
+                .map(|method| format!("- `{method}`", method = method))
+                .collect::<Vec<String>>();
+
+            let report = format!(
+                r####"# Forest API Implementation Report
```

Comment:
Maybe we can kick this out to a template text file? Seems out of place to be kept here as a magic string

Reply:
I'm not sure the format macro would even work that way, would it?

Reply:
That is a good question, I'm going to check.

Reply:
Yup, disregard. The macro requires a string literal

Reply:
Haha, macros continue to disappoint. Format, we still love you~